File: codeforces/2050/D_Digital_string_maximization.py
ntc = int(input())
for _ in range(ntc):
    inp = list(map(int, input()))
    i = len(inp) - 1
    # flag=True

    # while flag:
    #     flag=False
    #     for i in range(len(inp)-1, 0, -1):
    #         if inp[i]-1>inp[i-1]:
    #             #swap
    #             inp[i], inp[i-1] = inp[i-1], inp[i]-1
    #             flag=True

    while i > 0:
        if inp[i] - 1 > inp[i - 1]:
            # swap
            inp[i], inp[i - 1] = inp[i - 1], inp[i] - 1
            if i < len(inp) - 1:
                i += 1
            flag=True
        else:
            i -= 1
    print("".join(map(str, inp)))

File: codeforces/2050/D_Digital_string_maximization.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
  int ntc;
  cin >> ntc;

  while (ntc--) {
    string s;
    cin >> s;
    vector<int> inp;
    for (char c : s) {
      inp.push_back(c - '0');
    }

    bool flag = true;
    while (flag) {
      flag = false;
      for (int i = inp.size() - 1; i > 0; i--) {
        if (inp[i] - 1 > inp[i - 1]) {
          // swap
          int temp = inp[i - 1];
          inp[i - 1] = inp[i] - 1;
          inp[i] = temp;
          flag = true;
        }
      }
    }

    for (int num : inp) {
      cout << num;
    }
    cout << endl;
  }

  return 0;
}
